bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/util/AbstractContainerImpl.java

```diff
@@ -68,24 +75,54 @@ public List<? extends ComponentExporter> getItems() {
 
     protected <T> List<T> getChildrenModels(@Nullable SlingHttpServletRequest request, @NotNull Class<T> modelClass) {
         List<T> models = new ArrayList<>();
-
-        for (Resource child : slingModelFilter.filterChildResources(resource.getChildren())) {
-            if (!child.getName().startsWith("fd:")) {
-                T model = null;
-                if (request != null) {
-                    // todo: if possible set i18n form parent to child here, this would optimize the first form rendering
-                    model = modelFactory.getModelFromWrappedRequest(request, child, modelClass);
-                } else {
-                    model = child.adaptTo(modelClass);
-                    if (model instanceof Base && i18n != null) {
-                        ((Base) model).setI18n(i18n);
-                    }
+        List<Resource> filteredChildrenResources = getFilteredChildrenResources();
+        for (Resource child : filteredChildrenResources) {
+            T model = null;
+            if (request != null) {
+                // todo: if possible set i18n form parent to child here, this would optimize the first form rendering
+                model = modelFactory.getModelFromWrappedRequest(request, child, modelClass);
+            } else {
+                model = child.adaptTo(modelClass);
+                if (model instanceof Base && i18n != null) {
+                    ((Base) model).setI18n(i18n);
                 }
-                if (model != null) {
-                    models.add(model);
+            }
+            if (model != null) {
+                models.add(model);
+            }
+        }
+        return models;
+    }
+
+    @Override
+    public @NotNull Map<String, ? extends ComponentExporter> getExportedItems() {
+        if (itemModels == null) {
+            itemModels = getItemModels(request, ComponentExporter.class);
+        }
+        return itemModels;
+    }
+
+    private List<Resource> getFilteredChildrenResources() {
+        if (filteredChildComponents == null) {
+            filteredChildComponents = new LinkedList<>();
+            for (Resource child : slingModelFilter.filterChildResources(resource.getChildren())) {
+                if (!child.getName().startsWith("fd:")) {
+                    filteredChildComponents.add(child);
                 }
             }
         }
+        return filteredChildComponents;
+    }
+
+    protected Map<String, ComponentExporter> getItemModels(@NotNull final SlingHttpServletRequest request,
+        @NotNull final Class<ComponentExporter> modelClass) {
+        Map<String, ComponentExporter> models = new LinkedHashMap<>();
+        getFilteredChildrenResources().forEach(child -> {
+            ComponentExporter model = modelFactory.getModelFromWrappedRequest(request, child, modelClass);
+            if (model != null) {
+                models.put(child.getPath(), model);
```

Comment:
If the key is name then again we won't get the path of the resource. The value of the map is model object, we don't have path there.

Reply:
I am a little confused here. The code was already ignoring the `fd:` nodes.
We have refactored that code only to use in exportedItems API. But why is it only for wizard this is needed ?

Reply:
Hi @vdua ,
 In case of other layout components like tabs and accordion, they extend from corresponding wcm components.
 
  In the sling model of wcm components we have exposed ListItem object( which is a wrapper for Sling Resource object in one of the implementations) from return type of getItems Api. So while rendering the child components of the other layouts we get the path of the child components from this api. 
  
  But in aem core components we have exposed only the sling models with no property containing  the path of the resource, so in case of wizard, while rendering the childrens using data-sly-resource, we are unable to pass the path of the children, even the getName api doesn't provide the node name, it provides the jcr property name which is configured via cq dialog.

Moving ahead, we will also use this same api for accordion, tabs on top etc. Removing two sling model imports from their htl.